source/except.d.ts

```diff
@@ -36,22 +45,70 @@ This type is a stricter version of [`Omit`](https://www.typescriptlang.org/docs/
 
 This type was proposed to the TypeScript team, which declined it, saying they prefer that libraries implement stricter versions of the built-in types ([microsoft/TypeScript#30825](https://github.com/microsoft/TypeScript/issues/30825#issuecomment-523668235)).
 
+The NonStrictExcept refers to the standard Except method that permits the assignment of an object with additional properties to an object of a different type.
+@see https://github.com/sindresorhus/type-fest/issues/556
+
+@category Object
+*/
+type NonStrictExcept<ObjectType, KeysType extends keyof ObjectType> = {
+	[KeyType in keyof ObjectType as Filter<KeyType, KeysType>]: ObjectType[KeyType];
+};
+
+/**
+Create a type from an object type without certain keys - in stricter way.
+When the "strict" option in ExceptOptions is true, StrictExcept will be used.
+
+The StrictExcept method resolves the problem that arises when an object with additional properties is assigned to an object of a different type.
+@see https://github.com/sindresorhus/type-fest/issues/556
+
+@category Object
+*/
+type StrictExcept<ObjectType, KeysType extends keyof ObjectType> = NonStrictExcept<ObjectType, KeysType> & Partial<Record<KeysType, never>>;
+
+/**
+The Except type is the exported type, which determines the appropriate method to use (NonStrictExcept or StrictExcept) based on the options provided as the third argument (which is set to false by default).
+
 @example
 ```
-import type {Except} from 'type-fest';
+import {Except} from 'type-fest';
 
 type Foo = {
 	a: number;
 	b: string;
 	c: boolean;
 };
 
-type FooWithoutA = Except<Foo, 'a' | 'c'>;
-//=> {b: string};
+type FooWithoutA = Except<Foo, 'a', {strict: false}>; // False by default
+
+const foo: Foo = {
+	a: 1,
+	b: 'b',
+	c: true,
+};
+
+const fooWithoutA: FooWithoutA = foo; // No error
+//=> NonStrictExcept<Foo, 'a'>;
+
+@example
 ```
+import {Except} from 'type-fest';
 
-@category Object
-*/
-export type Except<ObjectType, KeysType extends keyof ObjectType> = {
-	[KeyType in keyof ObjectType as Filter<KeyType, KeysType>]: ObjectType[KeyType];
+type Foo = {
+	a: number;
+	b: string;
+	c: boolean;
+};
+
+type FooWithoutA = Except<Foo, 'a', {strict: true}>;
+
+const foo: Foo = {
+	a: 1,
+	b: 'b',
+	c: true,
 };
+
+const fooWithoutA: FooWithoutA = foo; // Error
+//=> StrictExcept<Foo, 'a'>;
+*/
+export type Except<ObjectType, KeysType extends keyof ObjectType, Options extends ExceptOptions = {strict: false}> =
+Options['strict'] extends false ? NonStrictExcept<ObjectType, KeysType> : StrictExcept<ObjectType, KeysType>;
```

Comment:
No need for extra types, we can just inline:
```suggestion
@example
```
import type {Except} from 'type-fest';

type Foo = {
	a: number;
	b: string;
};

type FooWithoutA = Except<Foo, 'a'>;
//=> {b: string}

const fooWithoutA: FooWithoutA = {a: 1, b: '2'};
//=> errors: 'a' does not exist in type '{ b: string; }'

type FooWithoutB = Except<Foo, 'b', {strict: true}>;
//=> {a: number} & Partial<Record<"b", never>>

const fooWithoutB: FooWithoutB = {a: 1, b: '2'};
//=> errors: Type 'string' is not assignable to type 'undefined'.
```

@category Object
*/
export type Except<ObjectType, KeysType extends keyof ObjectType, Options extends ExceptOptions = {strict: false}> = {
	[KeyType in keyof ObjectType as Filter<KeyType, KeysType>]: ObjectType[KeyType];
} & (Options['strict'] extends true
	? Partial<Record<KeysType, never>>
	: {});
```

Reply:
Wouldn't allow me to make a suggestion there, but in the `Except` doc comment:

```diff
/**
Create a type from an object type without certain keys.

+ Use option `strict: true` to disallow assigning additional properties to this type. 

// ...
*/
```